### PR TITLE
Update the solvers parameters to reflect the results of the paper and some small fixes.

### PR DESCRIPTION
**Pull request by FarhadK-1QBit**
### Description

Updated the parameters of the `dl-ccvm`, `mf-ccvm`, and `langevin` solvers to improve performance and to reflect the results of our recent research.
Fixed an issue with the storing of the `solve_time` and `pp_time` in all solvers.

### References
Closes issue #172 

### Testing
Can be tested by checking the results of running the examples files to see if the success probabilities (e.g. `optimal`, `one_percent`, etc.) are nonzero.

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: ccvm_simulators/solvers/dl_solver.py [hunk 1]
```diff
@@ -666,8 +675,8 @@ def __call__(
                 f"Solver option type {type(algorithm_parameters)} is not supported."
             )
 
-        # Stop the timer for the solve
-        solve_time = time.time() - solve_time_start
+        # Stop the timer for the solve to compute the solution time for solving an instance once
+        solve_time = (time.time() - solve_time_start) / batch_size
```
[the remark this answers is not in the document]
Comment (KatieO4t1qbit) [reply ccvm_simulators/solvers/dl_solver.py:679]: It would also be helpful to clarify what solve_time represents in the other places we have documented it:
- [in Solution](https://github.com/1QB-Information-Technologies/ccvm/blob/dev/solvers_parameters_update/ccvm_simulators/solution.py#L18)
- [in README](https://github.com/1QB-Information-Technologies/ccvm/blob/dev/solvers_parameters_update/README.md?plain=1#L125)

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,12 +10,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-### Fixed 
+### Fixed
 - Fixed issue where `s` was being updated incorrectly on each iteration  of `DLSolver._solve()`.
+- Fixed the calculation of `solve_time` and `pp_time` for all solvers to reflect the time
+for a single calculation of an instance intead of the time to solve the full batch
 
 ### Added
 - Added `PumpedLangevinSolver`, which is an extension of `LangevinSolver` to simulate pumped Langevin dynamics with a demo script in the examples directory.
 - Implemented a simple gradient descent post-processing step, as described in the paper, similar to Langevin dynamics but without noise; uses the Euler method with box constraint imposition at each iteration.
+- Added a saturation parameter `S` as the input to the `langevin_solver`
```
Comment (KatieO4t1qbit) [inline CHANGELOG.md:21]: It looks like S was also removed from the `__init__` function, which makes this a breaking change. Can you please include that information and move this point up to the breaking changes section?

Comment (KatieO4t1qbit) [reply CHANGELOG.md:21]: Moreover, I think we should separate this change into a separate PR, so that when it is merged the commit logs will more clearly show which commit introduced this breaking change just by the name.

**Review by KatieO4t1qbit — CHANGES_REQUESTED** [review]
Thanks for the PR, Farhad! 🎉 
My biggest concern is that there's a small part of this change that breaks the interface, so I've included some suggestions on how we can improve our inclusion of that change. The rest is looking good 😎

File: ccvm_simulators/solvers/dl_solver.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (FarhadK-1QBit) [reply ccvm_simulators/solvers/dl_solver.py:679]: Very good point @KatieO4t1qbit , and thanks for your suggestion. This is something we have extensively discussed and have come to the conclusion that this should be the way we measure solving time when using `gpu`. We are comparing different hardware and the advantage of a GPU lies in doing the parallelization. Since GPU can solve multiple instances simultaneously, it is faster to use to solve an instance once compare to CPU. 

I will add a line to the comment above this line and update the documentations in `Solution` and `README`. Thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (FarhadK-1QBit) [reply CHANGELOG.md:21]: Sounds good! I will create a new PR and issue for this change and log in the breaking changes section.

File: examples/ccvm_boxqp_mf.py [hunk 3]
```diff
@@ -15,12 +15,12 @@
...
-            "feedback_scale": 20,
-            "j": 20,
-            "S": 0.2,
+            "pump": 0.0,
+            "feedback_scale": 4000,
+            "j": 5.0,
+            "S": 20.0,
             "dt": 0.0025,
-            "iterations": 15000,
+            "iterations": 1500,
```
Comment (MehmetC-1QBit) [inline examples/ccvm_boxqp_mf.py:23]: I assumed the change in values of some of the parameters is to reflect the change in the paper, right?

**Review by MehmetC-1QBit — APPROVED** [review]
Thanks Farhad I approve the PR. But I have couple of comments regarding the change in default parameter setting just for curiosity. You don't have to necessarily answer them here:)

File: ccvm_simulators/solvers/dl_solver.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KatieO4t1qbit) [reply ccvm_simulators/solvers/dl_solver.py:682]: Thanks, Farhad! Looks good.

File: ccvm_simulators/solvers/dl_solver.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 import torch.distributions as tdist
 import time
 
-DL_SCALING_MULTIPLIER = 0.5
+DL_SCALING_MULTIPLIER = 0.2
```
[the remark this answers is not in the document]
Comment (FarhadK-1QBit) [reply ccvm_simulators/solvers/dl_solver.py:10]: I have performed some parameter tuning since the first time I set these parameters and have found that for the newly generated random instances these parameters work better. Same goes for the other solvers.

File: ccvm_simulators/solvers/mf_solver.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 import torch.distributions as tdist
 import time
 
-MF_SCALING_MULTIPLIER = 0.1
+MF_SCALING_MULTIPLIER = 0.05
```
[the remark this answers is not in the document]
Comment (FarhadK-1QBit) [reply ccvm_simulators/solvers/mf_solver.py:10]: Please see my response above.

File: examples/ccvm_boxqp_mf.py (hunk 3 above)
Comment (FarhadK-1QBit) [reply examples/ccvm_boxqp_mf.py:23]: Yes, that's correct! The large value of 15000 is no longer necessary as the solvers work find with only 1500 iterations.

File: ccvm_simulators/solvers/dl_solver.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MehmetC-1QBit) [reply ccvm_simulators/solvers/dl_solver.py:10]: That's awesome, thanks.

**Review by KatieO4t1qbit — APPROVED** [review]
This looks fantastic, thanks Farhad :)
